### ISSUE TITLE
Slicing removes styling

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#423]
```python
plot = (hv.Curve(np.arange(0, 10))(style={'linestyle':'--', 'color':'r'}) 
 * hv.Curve(np.arange(-10, 0))(style={'linestyle':'--', 'color':'b'}))
plot
```
<img width="272" alt="screen shot 2016-01-28 at 15 09 02" src="https://cloud.githubusercontent.com/assets/6897215/12646568/27e79d48-c5d1-11e5-923c-810f93262172.png">

```python
plot[:, -5:5]
```
<img width="281" alt="screen shot 2016-01-28 at 15 09 05" src="https://cloud.githubusercontent.com/assets/6897215/12646569/2a390b22-c5d1-11e5-9ea9-14eedf6c538a.png">

I'm on latest dev version

**Comment by jlstevens** [comment]
I've seen this behavior before when the id is not propagated after slicing. Should be an easy fix (I think it was fixed before but may have got lost in the data API refactor).

**Comment by jlstevens** [comment]
I can reproduce this issue and as I expected, the ``id`` values are incorrect after slicing. Should be easy to fix.

**Comment by philippjfr** [comment]
Fixed in #439. Styling is now always retained when cloning an object which is what's used by most internal operations when creating a new object. Let me know if you run into any issues. Closing now.
